docs/source/dev_newbuildpack.md

```diff
@@ -0,0 +1,27 @@
+# Adding a new buildpack to repo2docker
+
+A new buildpack is needed when a new language or a new package manager should be
+supported. Buildpacks need to work together. For example, when adding support for
+a new language, like nodejs, the new nodejs buildpack should be able to co-exist
+with Python / R / etc support.
```

Comment:
I think saying that build packs should co-exists will confuse users/make them believe multiple build packs can be applied at the same time. At least I believed that for a long time. Currently we try build packs in turn until we find one that says "yep, I match" then we use that one and stop trying others.

We started discussing this [here](https://github.com/jupyter/repo2docker/issues/230#issuecomment-364734365).